src/nepiada/utils/agent.py

```diff
@@ -16,10 +16,10 @@ def __init__(self):
         self.p_vel = None
 
 class Agent(Entity):  # properties of agent entities
-    def __init__(self, type):
+    def __init__(self, type, uid):
```

Comment:
This could present a case where two agents might have the same uid. I don't like this method of assigning IDs.

Reply:
We should add some checks around it to ensure that uid are unique

Reply:
We theoretically could use something like itertools, or just declare a global variable that we increment each time. Though I don't think it's necessary to implement such a check at the class level, I think typically good practice would be to enforce it where the agents are created. What do you think?

Reply:
Never mind, investigated it and made it global.